x-pack/plugins/security_solution/docs/prebuilt_rules_customization_rfc.md

```diff
@@ -0,0 +1,2397 @@
+    -   [Rule Management endpoints](#rule-management-endpoints)
+    -   [Prebuilt Rules endpoints](#prebuilt-rules-endpoints)
+    -   [Rule monitoring endpoints](#rule-monitoring-endpoints)
+    -   [Rule Execution Logs](#rule-execution-logs)
+-   [Exporting and importing rules](#exporting-and-importing-rules)
+    -   [Exporting rules](#exporting-rules)
+    -   [Importing rules](#importing-rules)
+    -   [Handling the `version` parameter](#handling-the-version-parameter)
+-   [Customizing Prebuilt Rules](#customizing-prebuilt-rules)
+    -   [Endpoints](#endpoints)
+        -   [Changes needed to endpoints](#changes-needed-to-endpoints)
+            -   [Update Rule - `PUT /rules`](#update-rule---put-rules)
+            -   [Patch Rule - `PATCH /rules`](#patch-rule---patch-rules)
+            -   [Bulk Patch Rules - `PATCH /rules/_bulk_update`](#bulk-patch-rules---patch-rules_bulk_update)
+            -   [Bulk Update Rules - `PUT /rules/_bulk_update`](#bulk-update-rules---put-rules_bulk_update)
+            -   [Bulk Actions - `POST /rules/_bulk_action`](#bulk-actions---post-rules_bulk_action)
+        -   [Updating the `customized` field](#updating-the-customized-field)
+    -   [In the UI](#in-the-ui)
+        -   [Via the Rule Edit Page](#via-the-rule-edit-page)
+        -   [Via Bulk Actions](#via-bulk-actions)
+        -   [Via the Rules Details Page](#via-the-rules-details-page)
+        -   [Via the Shared Exception Lists page](#via-the-shared-exception-lists-page)
+        -   [Via the Stack Management \> Rules UI](#via-the-stack-management-rules-ui)
+-   [List of things to fix (create tickets)](#list-of-things-to-fix-create-tickets)
+    -   [Rule fields](#rule-fields)
+-   [Updating Prebuilt Rules](#updating-prebuilt-rules)
+    -   [Changes to upgrade `_review` and `_perform` endpoints](#changes-to-upgrade-_review-and-_perform-endpoints)
+    -   [Rule field diff algorithms](#rule-field-diff-algorithms)
+-   [Changes in UI](#changes-in-ui)
+-   [Scenarios for bulk accepting updates](#scenarios-for-bulk-accepting-updates)
+-   [Marking rules (and their fields) as customized](#marking-rules-and-their-fields-as-customized)
+-   [Other open questions](#other-open-questions)
+
+## Necessary rule schema changes
+
+In order to support the customization of Elastic Prebuilt Rules, we need to modify our rule schema. This involves introducing a new nested `prebuilt` field and deprecating the `immutable` field.
+
+### `prebuilt` and `immutable` fields
+
+The `prebuilt` field will be a top-level object field in our rule schema that will contain subfields with information relating to Elastic Prebuilt rules.
+
+It will be an optional field, which will only be present for Elastic prebuilt rules.
+
+Its existence within a rule object will determine if a rule is a prebuilt rule (and its absence will determine that it is a custom rule).
```

Comment:
Yes, this is an issue. I described a workaround below but have very big doubts about it.

Reply:
I think this is fine, but it could make more sense if we have a field inside `prebuilt` (or the renamed equivalent, see my other [comment](https://github.com/elastic/kibana/pull/171856/files#r1482151547) like a `repo_name` or `author` that is intuitively required and very commonly used as a filter ("give me all prebuilt rules from `Elastic`"). Then "give me all prebuilt rules" is equivalent to "give me all prebuilt rules from any repo" - `prebuilt.repo_name: *`.